ndc-reference/bin/reference/main.rs

```diff
@@ -1528,52 +1634,148 @@ fn eval_aggregate_function(
     function: &models::AggregateFunctionName,
     values: &[serde_json::Value],
 ) -> Result<serde_json::Value> {
-    let int_values = values
-        .iter()
-        .map(|value| {
-            value
-                .as_i64()
-                .ok_or((
-                    StatusCode::BAD_REQUEST,
-                    Json(models::ErrorResponse {
-                        message: "column is not an integer".into(),
-                        details: serde_json::Value::Null,
-                    }),
-                ))?
-                .try_into()
-                .map_err(|_| {
-                    (
-                        StatusCode::BAD_REQUEST,
-                        Json(models::ErrorResponse {
-                            message: "column value out of range".into(),
-                            details: serde_json::Value::Null,
-                        }),
-                    )
+    if let Some(first_value) = values.iter().next() {
```

Comment:
This function is quoted verbatim in the spec, so it's probably going to be a bit hard to follow now. You might want to break up the code a bit in the text.

Reply:
OK, I've reduced the example in the docs down to just the first part of the function. The rest are basically the same just slight variants based on the type. The function isn't actually that big, it's just the shitty rust formatter that loves adding as many newlines as it possibly can blowing out every piece of error handling code into a lot of lines.